### ISSUE TITLE
Support for testing a Kafka Connect connector

### ISSUE THREAD

**Issue by rjrudin** [mguenther/kafka-junit#71]
This is a question, not a bug - I'm working on a custom Kafka Connector connector (currently, just a sink connector) and have been wondering how I could test it. I found your project and scanned through the docs at https://mguenther.github.io/kafka-junit/ , but I didn't see any mention of "connector". My anticipation is I'd do something like this in pseudocode:

- Configure and start an EmbeddedKafkaCluster
- Add/register my custom connector class
- Configure my connector class to listen to a particular topic
- Send test messages to that topic
- Verify that my connector receives and processes those messages correctly

Does kafka-junit provide support for that?

**Comment by mguenther** [comment]
Yes, this is possible. Kafka for JUnit allows you to deploy a Kafka Connect worker. Unfortunately, the documentation is lacking a simple example (which I'll need to address in the future, thanks for pointing it out!). I'd recommend a look into the suite of integration tests for Kafka for Junit. There you'll find the `ConnectorTest` class that shows you how to deploy a Kafka Connect worker. But please keep in mind that your Kafka Connect worker is a black-box to Kafka for JUnit. This is especially relevant for sink connectors that target systems other than Kafka, as Kafka for Junit cannot provide any form of instrumentation for those systems nor can it spy on the actual logic of your Kafka Connect worker.

Hope this helps.

**Comment by rjrudin** [comment]
Thanks @mguenther - tests are really good documentation! I took a look at [ConnectorTest](https://github.com/mguenther/kafka-junit/blob/development/src/test/java/net/mguenther/kafka/junit/ConnectorTest.java) and the associated connector, that looks like a good starting point. And having Kafka for Junit treat a sink connector as a black box is perfectly fine. 

I'll close this ticket as being addressed, will open a new one if I run into any issues but I think this is what I needed to get started.
